diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/BooleanExpressionsRule.kt

```diff
@@ -0,0 +1,165 @@
+import org.cqfn.diktat.ruleset.utils.KotlinParser
+import org.cqfn.diktat.ruleset.utils.findAllNodesWithCondition
+import org.cqfn.diktat.ruleset.utils.findLeafWithSpecificType
+
+import com.bpodgursky.jbool_expressions.Expression
+import com.bpodgursky.jbool_expressions.parsers.ExprParser
+import com.bpodgursky.jbool_expressions.rules.RuleSet
+import com.pinterest.ktlint.core.ast.ElementType
+import com.pinterest.ktlint.core.ast.ElementType.BINARY_EXPRESSION
+import com.pinterest.ktlint.core.ast.ElementType.OPERATION_REFERENCE
+import com.pinterest.ktlint.core.ast.ElementType.PARENTHESIZED
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+
+import java.lang.RuntimeException
+
+/**
+ * Rule that warns if the boolean expression can be simplified.
+ */
+class BooleanExpressionsRule(configRules: List<RulesConfig>) : DiktatRule(
+    "boolean-expressions-rule",
+    configRules,
+    listOf(COMPLEX_BOOLEAN_EXPRESSION)) {
+    override fun logic(node: ASTNode) {
+        if (node.elementType == ElementType.CONDITION) {
+            checkBooleanExpression(node)
+        }
+    }
+
+    @Suppress("TooGenericExceptionCaught")
+    private fun checkBooleanExpression(node: ASTNode) {
+        // This map is used to represent expressions as single chars, which are used by the lib.
+        val mapOfExpressionToChar: HashMap<String, Char> = HashMap()
+        val correctedExpression = makeCorrectExpressionString(node, mapOfExpressionToChar)
+        // If there are method calls in conditions
+        val expr: Expression<String> = try {
+            ExprParser.parse(correctedExpression)
+        } catch (runTimeExc: RuntimeException) {
+            return
+        }
+        val distributiveLawString = checkDistributiveLaw(expr, mapOfExpressionToChar, node)
+        val simplifiedExpression = distributiveLawString?.let {
+            ExprParser.parse(distributiveLawString)
+        }
+            ?: RuleSet.simplify(expr)
+        if (expr != simplifiedExpression) {
+            COMPLEX_BOOLEAN_EXPRESSION.warnAndFix(configRules, emitWarn, isFixMode, node.text, node.startOffset, node) {
+                fixBooleanExpression(node, simplifiedExpression, mapOfExpressionToChar)
+            }
+        }
+    }
+
+    private fun makeCorrectExpressionString(node: ASTNode, mapOfExpressionToChar: HashMap<String, Char>): String {
+        // `A` character in ASCII
+        var characterAsciiCode = 'A'.code
+        node
+            .findAllNodesWithCondition({ it.elementType == BINARY_EXPRESSION })
+            .filterNot { it.text.contains("&&") || it.text.contains("||") }
+            .forEach {
+                if (mapOfExpressionToChar.containsKey(it.text)) {
+                    return@forEach
+                }
+                // Assigning the boolean expression to a char. The lib represents expressions[a > 5] as chars [A, B, C ...].
+                mapOfExpressionToChar[it.text] = characterAsciiCode.toChar()
+                characterAsciiCode++
```

Comment:
This block can be simplified with `computeIfAbsent`

Reply:
Fixed